src/components/text_input/text_input.ts

```diff
@@ -64,11 +64,44 @@ export class TextInput extends Component<Props, SpreadsheetChildEnv> {
     );
   }
 
+  onKeyDown(ev: KeyboardEvent) {
+    switch (ev.key) {
+      case "Enter":
+        this.save();
+        this.inputRef.el!.blur();
+        ev.preventDefault();
+        ev.stopPropagation();
+        break;
+      case "Escape":
+        this.inputRef.el!.value = this.props.value;
+        this.inputRef.el!.blur();
+        ev.preventDefault();
+        ev.stopPropagation();
+        break;
+    }
+  }
+
   onChange() {
     this.save();
   }
 
   save() {
     this.props.onChange((this.inputRef.el?.value || "").trim());
   }
+
+  onFocus() {
+    this.focusInputAndSelectContent();
+  }
+
+  onBlur() {
+    this.save();
+  }
```

Comment:
can probably remove those methods and directly call `focusInputAndSelectContent` and `save`